### PR TITLE
[RLlib] RLTrainer stand-alone unittests

### PR DESCRIPTION
**Pull request by kouroshHakha**
## Why are these changes needed?
This PR implements the stand-alone unittests for RLTrainer. Right now it only includes tf stuff, later we will extend it to torch.
This follows up this https://github.com/ray-project/ray/pull/31511

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: rllib/core/rl_trainer/tests/tf/test_tf_rl_trainer.py [hunk 1]
```diff
@@ -23,6 +25,7 @@ def setUp(cls) -> None:
     def tearDown(cls) -> None:
         ray.shutdown()
 
+    @pytest.mark.skip
```
Comment (kouroshHakha) [inline rllib/core/rl_trainer/tests/tf/test_tf_rl_trainer.py:28]: We'll reactivate this test as a trainer_runner test once we iterate through the interface of trainer_runner and algorithm to solidify its design.

**Review by gjoliver — APPROVED** [review]

File: rllib/core/rl_trainer/tests/test_rl_trainer.py [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+        """Tests the compute_gradients correctness.
+
+        Tests that if we sum all the trainable variables the gradient of output w.r.t.
+        the weights is all ones.
+        """
+        trainer = get_trainer()
+
+        with tf.GradientTape() as tape:
+            params = trainer.module[DEFAULT_POLICY_ID].trainable_variables
+            loss = {"total_loss": sum([tf.reduce_sum(param) for param in params])}
```
Comment (avnishn) [inline rllib/core/rl_trainer/tests/test_rl_trainer.py:81]: nice property that you've used here.

File: rllib/utils/test_utils.py [hunk 3]
```diff
@@ -1146,7 +1146,7 @@ def get_cartpole_dataset_reader(batch_size: int = 1) -> "DatasetReader":
         get_dataset_and_shards,
     )
 
-    path = "tests/data/cartpole/large.json"
+    path = "rllib/tests/data/cartpole/large.json"
```
Comment (avnishn) [inline rllib/utils/test_utils.py]: I think you need to remove this prepend

**Review by avnishn — APPROVED** [review]

File: rllib/utils/test_utils.py (hunk 3 above)
Comment (kouroshHakha) [reply rllib/utils/test_utils.py]: yep yep. just pushed a patch.

**Pull request merged**